com.unity.ml-agents/Runtime/Grpc/RpcCommunicator.cs

```diff
@@ -504,7 +504,7 @@ public static byte[] GetSideChannelMessage(Dictionary<int, SideChannel> sideChan
                         var messageList = sideChannel.MessageQueue;
                         foreach (var message in messageList)
                         {
-                            binaryWriter.Write(sideChannel.ChannelType());
+                            binaryWriter.Write(sideChannel.ChannelId.ToByteArray());
```

Comment:
This approach might not be fully correct according to this: https://stackoverflow.com/a/16722909/224264

Not sure if it's worth fixing though.

Reply:
The fix is on the Python side. We read guid with [bytes_le](https://docs.python.org/2/library/uuid.html#uuid.UUID.bytes_le) argument.